lms/djangoapps/courseware/grades.py

```diff
@@ -499,10 +559,13 @@ def _progress_summary(student, course, course_structure=None):
         course_structure = get_course_blocks(student, course.location)
     if not len(course_structure):
         return None
-    scorable_locations = [block_key for block_key in course_structure if possibly_scored(block_key)]
-
-    with outer_atomic():
-        scores_client = ScoresClient.create_for_locations(course.id, student.id, scorable_locations)
+    if scores_client is None:
+        if grading_context_result is None:
+            scorable_locations = [block_key for block_key in course_structure if possibly_scored(block_key)]
+        else:  # TODO: these two list are equivalent, yeah?
```

Comment:
@nasthagiri I think setting up the scores client with extraneous locations will be a safe switch. Here's my logic:

- The smaller set of scorable locations is only used by `_grade()`, and only as a parameter to `_calculate_totaled_scores()`
  - for reference, it's calculated as `[block.location for block in grading_context_result['all_graded_blocks']]`
  - also, this is the only usage of `_calculate_totaled_scores()`
- In that method, `scores_client` is used in 2 places:
  - as a call to `get_score()`, which is the pattern @ormsbee describes above. Extra locations in the pool should not affect the ability to get a score for a given location.
  - as a test in the following loop: https://github.com/edx/edx-platform/blob/b6ad020879dd3d5f37b81fae98ac4889897546b0/lms/djangoapps/courseware/grades.py#L423-L426
    - this would be a problem, except if we look at where `section_info['scored_descendants']` and the location list used to initialize the scores client are defined, it looks like the list we're iterating over is already a subset of the locations used to initialize `scores_client`
    - https://github.com/edx/edx-platform/blob/b6ad020879dd3d5f37b81fae98ac4889897546b0/lms/djangoapps/courseware/grades.py#L156-L165
    - Because of this, `section_info['scored_descendants']` is the limiting set, and expanding the size of the scores_client locations can occur safely.